kernel/kernel-api/src/main/java/io/delta/kernel/engine/Engine.java

```diff
@@ -56,6 +57,23 @@ public interface Engine {
    */
   ParquetHandler getParquetHandler();
 
+  /**
+   * Retrieves a {@link CommitCoordinatorClient} for the specified commit coordinator name.
+   *
+   * @param commitCoordinatorName The name (identifier) of the underlying commit coordinator client
+   *     to instantiate
+   * @param commitCoordinatorConf The configuration settings for the underlying commit coordinator
+   *     client, taken directly from the Delta table property {@link
+   *     io.delta.kernel.internal.TableConfig#COORDINATED_COMMITS_TABLE_CONF}
+   * @return A {@link CommitCoordinatorClient} implementation corresponding to the specified commit
+   *     coordinator name
+   * @since 3.3.0
+   */
+  default CommitCoordinatorClient getCommitCoordinatorClient(
+      String commitCoordinatorName, Map<String, String> commitCoordinatorConf) {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
Is this temporarily with a default implementation or will we be keeping it like this?

What will be the expected behavior if an engine interface hasn't implemented this method but some user tries to read a CC table? Will it throw this exception? Or do we want to force all engine impls to override this?

Reply:
> temporarily with a default implementation
Yup! This is so we can merge this without having to go and update all implementations of Engine within this PR

Reply:
Is it right to assume that implementations will still be able to read other dynamic configurations when building the coordinator? e.g. the Delta-spark getCCC interface also takes in a sparkSession allowing for dynamic configuration of the client. Implementations of this method will still be able to read some other configuration source (even though it is not explicitly being passed) right?

Reply:
> Is it right to assume that implementations will still be able to read other dynamic configurations when building the coordinator?

Absolutely. We leave it up to the engine to create the CCC. If the engine is aware of any dynamodb configurations, it can use them!

> Implementations of this method will still be able to read some other configuration source (even though it is not explicitly being passed) right?

Yes. I'd encourage you to look at the tracking issue https://github.com/delta-io/delta/issues/3817 to look at future PRs where you can see this being done.